doc/specs/stdlib_version.md

```diff
@@ -0,0 +1,64 @@
+---
+title: Version information
+---
+
+# The `stdlib_version` module
+
+[TOC]
+
+## Introduction
+
+The `stdlib_version` module contains the version of the standard library.
+The version information can be used as compile time constant or retrieved from a getter function at runtime.
+In case the standard library is dynamically linked, the version number retrieved from the getter might mismatch the compile time constants provided from the version built against.
+Therefore, it is recommended to retrieve the version information always at runtime.
```

Comment:
Why not only allow the correct approach to do it?

Reply:
For the compact version I can see an application to verify the compatibility of API and/or ABI at runtime by a construct like:
```f90
use stdlib_version, only : stdlib_version_compact, get_stdlib_version
implicit none
integer :: major
call get_stdlib_version(major=major)
if (major > stdlib_version_compact / 10000) then
  error stop "Incompatible version of stdlib used"
end if
end
```

Similarly, `stdlib_version_string` and `get_stdlib_version(string=string)` can be used to provide a printout to inform about the stdlib versions used at build time and run time.

Not sure whether this is a realistic application of this information.

Reply:
Ok. Thank you for the explanation. You can merge this PR from my point-of-view.